src/ApiCompat/Microsoft.DotNet.ApiCompatibility/Resources.resx

```diff
@@ -142,10 +142,10 @@
     <value>{2} assembly version '{0}' should be equal to or higher than {3} version '{1}'.</value>
   </data>
   <data name="ElementShouldNotBeNullAtIndex" xml:space="preserve">
-    <value>Element should not be null. Found null at index: {0}.</value>
+    <value>Element at index {0} should not be null.</value>
   </data>
   <data name="IndexShouldBeWithinSetSizeRange" xml:space="preserve">
-    <value>The index should be within the right set size range.</value>
+    <value>The index should be within the right set size range (0, {0}).</value>
```

Comment:
The range isn't well specified here as the first value is legal but not the last value. Suggest using the value of length - 1 so the message can say should be between zero and that inclusive.

Reply:
The implementation says

Reply:
I included nameof(Right) and added to the string
`The {0} index should be in the range zero through {1} inclusive.`

Reply:
Yeah, I know.
I'm just trying to push myself into the Open Source world hahaha ❤